python/datafusion/dataframe.py

```diff
@@ -160,6 +160,53 @@ def with_column(self, name: str, expr: Expr) -> DataFrame:
         """
         return DataFrame(self.df.with_column(name, expr.expr))
 
+    def with_columns(
+        self, *exprs: Expr | Iterable[Expr], **named_exprs: Expr
+    ) -> DataFrame:
+        """Add columns to the DataFrame.
+
+        By passing expressions, iteratables of expressions, or named expressions. To
+        pass named expressions use the form name=Expr.
+
+        Example usage:
+
+            The following will add 4 columns labeled a, b, c, and d.
+
+            df = df.with_columns(
+                lit(0).alias('a'),
+                [lit(1).alias('b'), lit(2).alias('c')],
+                d=lit(3)
+                )
+
+        Args:
+            *exprs: Name of the column to add.
+            **named_exprs: Expression to compute the column.
+
+        Returns:
+            DataFrame with the new column.
+        """
```

Comment:
```suggestion
    def with_columns(
        self, *exprs: Expr | Iterable[Expr], **named_exprs: Expr
    ) -> DataFrame:
        """Add columns to the DataFrame.

        By passing expressions, iteratables of expressions, or named expressions. To
        pass named expressions use the form name=Expr.

        Example usage: The following will add 4 columns labeled a, b, c, and d::

            df = df.with_columns(
                lit(0).alias('a'),
                [lit(1).alias('b'), lit(2).alias('c')],
                d=lit(3)
                )

        Args:
            exprs: Either a single expression or an iterable of expressions to add.
            named_exprs: Named expressions in the form of ``name=expr``

        Returns:
            DataFrame with the new columns added.
        """
```

Reply:
Sorry for the multiple review comments. The above code renders correctly in the docs when I generate them